test/move_effect_healing_wish.c

```diff
@@ -0,0 +1,77 @@
+#include "global.h"
+#include "test_battle.h"
+
+ASSUMPTIONS
+{
+    ASSUME(gBattleMoves[MOVE_HEALING_WISH].effect == EFFECT_HEALING_WISH);
+    ASSUME(gBattleMoves[MOVE_LUNAR_DANCE].effect == EFFECT_HEALING_WISH);
+}
+
+#define TEST_MAX_HP (100)
+
+SINGLE_BATTLE_TEST("Healing Wish causes the user to faint and fully heals the replacement")
+{
+    GIVEN {
+        ASSUME(B_HEALING_WISH_SWITCH >= GEN_5);
+        PLAYER(SPECIES_GARDEVOIR); // Faster than Wobbufeet.
+        PLAYER(SPECIES_WYNAUT) {HP(1) ; MaxHP(TEST_MAX_HP); Status1(STATUS1_POISON); };
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_HEALING_WISH); SEND_OUT(player, 1) ; }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_HEALING_WISH, player);
+        HP_BAR(player, hp: 0);
+        MESSAGE("Gardevoir fainted!");
+        MESSAGE("The healing wish came true for Wynaut!");
+        HP_BAR(player, hp: TEST_MAX_HP);
+        NOT STATUS_ICON(player, poison: TRUE);
```

Comment:
I think you mean `STATUS_ICON(player, none: TRUE);`.

`NOT STATUS_ICON(player, poison: TRUE);` says "the status icon won't change to be poison", whereas `STATUS_ICON(player, none: TRUE);` says "the status icon will change to nothing" (i.e. the status will be cured).